svcomp/src/main/java/com/dat3m/svcomp/options/SVCOMPOptions.java

```diff
@@ -75,14 +73,27 @@ public SVCOMPOptions(){
 
         addOption(new Option(BOOGIESAN, false,
                 "Generates (also) a sanitised boogie file saved as /output/X-sanitised.bpl"));
-}
+    }
     
-    public void parse(String[] args) throws ParseException, RuntimeException {
-    	super.parse(args);
-        if(supported_formats.stream().map(f -> programFilePath.endsWith(f)). allMatch(b -> b.equals(false))) {
-            throw new RuntimeException("Unrecognized program format");
+    public static SVCOMPOptions fromArgs(String[] args) {
+    	SVCOMPOptions options = new SVCOMPOptions();
+        try {
+            options.parse(args);
         }
-
+        catch (Exception e){
+            if(e instanceof UnsupportedOperationException){
+                System.out.println(e.getMessage());
+            }
+            new HelpFormatter().printHelp("SVCOMP Runner", options);
+            System.exit(1);
```

Comment:
I don't like that this method terminates the whole program on failure. I would rather let the caller handle the failure and terminate if necessary. I think forced termination is something that should happen as close to the entry point of the program as possible.

I think it should be like this: 
- If the class is completely public, you should make as few assumptions about the calling environment as possible. In this case, printing help messages and terminating is something that not every caller would expect. For example, it is reasonable to expect a caller to parse something and on failure to fall back onto e.g. standard options. This is something that the current design forbids.
- If you expect a certain calling context, then the visibility of this class or at least its constructing methods should be as limited as possible. This means you should either have them package private w.r.t. to the caller or even make this class an inner class of the caller if there is only one such caller (e.g. if only `Dartagnan` uses `DartagnanOptions` and only `SvcompRunner` uses `SvcompIOptions`). Then you can incorporate all this special behaviour on failure as you like.
If this scoping is not possible for some reason, then we should at least document the expected callers via comments.

I think you did some of this parsing in `ApplicationTests` as well and I believe you would not want to have printed messages or forced termination in those cases. I think `System.exit()` tries to terminate the whole JVM, so you wouldn't be able to add unit tests for wrong parsing (it would terminate the whole test suit I believe).

Reply:
I think the best would be letting `parse()` throw an exception and catch it in the `Dartagnan` main class. Then the catch can print the help.

Reply:
The first option is the cleanest one as it has the least assumptions.

Reply:
Printing the help is still done in the `parse()` for two reasons:
- `HelpFormatter().printHelp()` needs access to the `XOption` object
- I think getting the options help is always useful when there was a parsing problem.

Overall, the handling of exceptions was a mess from the application level. I tried to clean it up. Overall, now none of the `main` methods throws exceptions anymore and always ends "graciously" (maybe with exit code 1). Different exceptions are now catch and a corresponding msg is displayed. When the problem comes from the verification, the msg is logged and either `TIMEOUT` or `ERROR` is printed; if the problem was parsing an options then the helper is printed; when the problem was due to parsing the program or the CAT model (missing file, parsing problem, compilation problem), the reason is printed.